synfig-core/src/synfig/valuenodes/valuenode_add.h

```diff
@@ -62,10 +62,12 @@ class ValueNode_Add : public LinkableValueNode
 	virtual String get_local_name()const;
 	static bool check_type(Type &type);
 	virtual Vocab get_children_vocab_vfunc()const;
+
+	//! Checks if it is possible to call get_inverse() for target_value at time t.
+	//! If so, return the link_index related to the return value provided by get_inverse()
+	virtual InvertibleStatus is_invertible(Time t, const ValueBase& target_value, int* link_index = nullptr) const;
 	//! Returns the modified Link to match the target value at time t
-	ValueBase get_inverse(Time t, const synfig::Real &target_value) const;
-	ValueBase get_inverse(Time t, const synfig::Angle &target_value) const;
-	ValueBase get_inverse(Time t, const synfig::Vector &target_value) const;
+	virtual ValueBase get_inverse(Time t, const synfig::ValueBase &target_value) const;
```

Comment:
Can you add a little description for which purposes `get_inverse` is used?

Reply:
It's on valuenode.h . Should I make it here too?

Reply:
Oh I did on every other ;) I'll do it now

Reply:
Oops. It was there. Man, I'm not thinking well right now XD
As for me, it's complicated to explain it :) Can you help me? It's the inverse of operator()(Time t).
Used when, for example, user drag a handle and Studio needs to know how can it change the value node parameters to match new position.

Reply:
Sorry again. I completely missed that part in `valuenode.h`